File: tags/start/Foundation/orbsupport/test/testorbsupport/java/src/cdmw/orbsupport/test/MainTest.java
/* ===================================================================== */
/*
 * This file is part of CARDAMOM (R) which is jointly developed by THALES 
 * and SELEX-SI. 
 * 
 * It is derivative work based on PERCO Copyright (C) THALES 2000-2003. 
 * All rights reserved.
 * 
 * CARDAMOM is free software; you can redistribute it and/or modify it under 
 * the terms of the GNU Library General Public License as published by the
 * Free Software Foundation; either version 2 of the License, or (at your 
 * option) any later version. 
 * 
 * CARDAMOM is distributed in the hope that it will be useful, but WITHOUT 
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or 
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Library General Public 
 * License for more details. 
 * 
 * You should have received a copy of the GNU Library General 
 * Public License along with CARDAMOM; see the file COPYING. If not, write to 
 * the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
*/
/* ===================================================================== */


package cdmw.orbsupport.test;

import cdmw.ossupport.OS;

public class MainTest {

    public static void main(String[] args) {
try{
        // Are we started as the CORBA server ?
        if ( OS.getOptionValue( args, "--server").equals("yes") ) {

            ServerProcess serverp = new ServerProcess();
            try {
                serverp.start( args );
            }
            catch(org.omg.CORBA.UserException ue) {
                ue.printStackTrace();
            }

        // Else, we are the main test driver
        } else {

              ORBSupportTest ost = new ORBSupportTest();
              ost.start();

        }
}catch (Exception ex){
ex.printStackTrace();
}
     }

    public static void writeToFile(String ior,String filename) {

        java.io.File file = new java.io.File(filename);
        String s;

        try {
            java.io.PrintWriter out = new java.io.PrintWriter(
                new java.io.BufferedWriter( new java.io.FileWriter(file) ) );
            out.println(ior);
            out.close();
        } catch(java.io.IOException e) {}

    }

    public static String readFromFile(String file) {

        try {
            java.io.BufferedReader in = new java.io.BufferedReader(
                new java.io.FileReader(file) );
            return in.readLine();
        }
        catch(java.io.IOException ioe) {
            return "";
        }
    }

}

File: FaultTolerance/ftmonitoring/test/testmutual/c++/src/main_test.cpp
/* ===================================================================== */
/*
 * This file is part of CARDAMOM (R) which is jointly developed by THALES
 * and SELEX-SI. It is derivative work based on PERCO Copyright (C) THALES
 * 2000-2003. All rights reserved.
 * 
 * Copyright (C) THALES 2004-2005. All rights reserved
 * 
 * CARDAMOM is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Library General Public License as published
 * by the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 * 
 * CARDAMOM is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Library General Public
 * License for more details.
 * 
 * You should have received a copy of the GNU Library General Public
 * License along with CARDAMOM; see the file COPYING. If not, write to the
 * Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
*/
/* ===================================================================== */

// Std include files
#include <iostream>
#include <sstream>
#include <string.h>

#include "testmutual/TestMonitoringLibrary.hpp"

int main (int argc, char* argv[])
{
    
    int ret_code = 0;
    /*
  ACE_TRY_NEW_ENV
    {
      CORBA::ORB_var orb =
        CORBA::ORB_init (argc, argv, "" ACE_ENV_ARG_PARAMETER);
      ACE_TRY_CHECK;

      CORBA::Object_var poa_object =
        orb->resolve_initial_references("RootPOA" ACE_ENV_ARG_PARAMETER);
      ACE_TRY_CHECK;

      PortableServer::POA_var root_poa =
        PortableServer::POA::_narrow (poa_object.in () ACE_ENV_ARG_PARAMETER);
      ACE_TRY_CHECK;

      if (CORBA::is_nil (root_poa.in ()))
        ACE_ERROR_RETURN ((LM_ERROR,
                           " (%P|%t) Panic: nil RootPOA\n"),
                          1);

      PortableServer::POAManager_var poa_manager =
        root_poa->the_POAManager (ACE_ENV_SINGLE_ARG_PARAMETER);
      ACE_TRY_CHECK;

      // Get the group IOR.
      CORBA::String_var ior = CORBA::string_dup (group_ior);
      CORBA::Object_var group1 =
        orb->string_to_object (ior.in () ACE_ENV_ARG_PARAMETER);
      ACE_TRY_CHECK;

      ACE_DEBUG ((LM_DEBUG, "(%P|%t) create_id_for_reference of "));
      ACE_DEBUG ((LM_DEBUG, group_ior));
      ACE_DEBUG ((LM_DEBUG, "\n"));

      try
      {
        PortableServer::ObjectId_var id =
          root_poa->create_id_for_reference (group1.in ()
                                             ACE_ENV_ARG_PARAMETER);
      }  
      catch (const CORBA::SystemException& ex)
      {
        cout << ex._info().c_str() << std::endl;
        throw;
      }
    }
  ACE_CATCHANY
    {
      ACE_PRINT_EXCEPTION (ACE_ANY_EXCEPTION,
                           "Exception caught:");
      return 1;
    }
  ACE_ENDTRY;
    */
    
    TestMonitoringLibrary myTestMonitoringLibrary(argc, argv);
    myTestMonitoringLibrary.start();
    
    return ret_code;
}

